### PR TITLE
NOISSUE - Update Nginx Docker Image

### PR DESCRIPTION
**Pull request by rodneyosodo**
Signed-off-by: rodneyosodo <socials@rodneyosodo.com>

### What does this do?
Updates nginx docker image from `nginx:1.20.0-alpine` to `nginx:1.23.3-alpine`.  In doing so, it fixes typo is `docker/ssl/Makefile` and also return error of type `ErrBearerKey` on publish request instead of `ErrBearerToken`

### Which issue(s) does this PR fix/relate to?
N/A

### List any changes that modify/break current functionality
No

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
No

### Notes
N/A

### CODE REVIEW

File: docker/ssl/authorization.js [hunk 1]
```diff
@@ -167,10 +167,12 @@ function parseCert(cert, key) {
...
-                return pair[1];
+                return "Thing " + pair[1].replace("\\", "");
             }
         }
     }
 
     return '';
 }
+
+export default {setKey,authenticate};
```
Comment (dborovcanin) [inline docker/ssl/authorization.js]: Add an empty line to the end of the file.

**Review by drasko — DISMISSED** [review]
LGTM

**Review by dborovcanin — CHANGES_REQUESTED** [review]
Can you also generate CA and server certs since the old ones are expired?
(You can use Makefile from `docker/ssl` with commands `make ca` and `make server_cert`).

File: docker/ssl/authorization.js [hunk 2]
```diff
@@ -167,10 +167,12 @@ function parseCert(cert, key) {
         for (var i = 0; i < pairs.length; i++) {
             var pair = pairs[i].split('=');
             if (pair[0].toUpperCase() == key) {
-                return pair[1];
+                return "Thing " + pair[1].replace("\\", "");
```
Comment (dborovcanin) [inline docker/ssl/authorization.js]: Please check if the MQTT adapter handles it correctly in both cases when a prefix is sent in the password and when it's not.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: docker/ssl/authorization.js (hunk 2 above)
Comment (rodneyosodo) [reply docker/ssl/authorization.js]: I have updated

File: docker/ssl/authorization.js [hunk 3]
```diff
@@ -29,8 +29,8 @@ function authenticate(s) {
 
         var pass = parsePackage(s, data);
 
-        if (!clientKey.length || pass !== clientKey) {
-            s.error('Cert CN (' + clientKey + ') does not match client password');
+        if (!clientKey.length || !clientKey.includes(pass) ) {
```
Comment (dborovcanin) [inline docker/ssl/authorization.js]: Is there an option to use it like `endsWith` which is more precise for ignoring prefixes?

**Review by dborovcanin — CHANGES_REQUESTED** [review]

**Review by dborovcanin — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**